app/base/handlers.js

```diff
@@ -40,16 +47,16 @@ exports.home = (req, res, next) => {
 };
 
 exports.tools = (req, res, next) => {
-  const { rstudio_is_deploying } = req.session;
+  const { is_deploying } = req.session;
   const ns = cls.getNamespace(config.continuation_locals.namespace);
-  req.session.rstudio_is_deploying = false;
+  req.session.is_deploying = {};
 
-  Deployment.list()
-    .then((tools) => {
+  Promise.all([Deployment.list(), Tool.list()])
```

Comment:
We could probably cache the response of `Tool.list()` as they're not likely to change often. But that obviously increases complexity.